src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1508,28 +1557,199 @@ abstract void onApplication(
         private final class MqttPublishStream extends MqttServerStream
         {
             private final MessageConsumer application;
+
             private long routeId;
             private long initialId;
             private long replyId;
-            private int packetId;
+
             private String topicFilter;
 
+            private int state;
+
+            private int packetId;
+
+            private int initialBudget;
+            private int initialPadding;
+            private int replyBudget;
+
+            private int initialSlot = NO_SLOT;
+            private int initialSlotOffset;
+            private long initialSlotTraceId;
+
+            private Future<?> signalFuture;
+
             MqttPublishStream(
                 MessageConsumer application,
                 long routeId,
                 long initialId,
                 long replyId,
-                int packetId)
+                int packetId,
+                String topicFilter)
             {
                 this.application = application;
                 this.routeId = routeId;
                 this.initialId = initialId;
                 this.replyId = replyId;
                 this.packetId = packetId;
+                this.topicFilter = topicFilter;
+            }
+
+            private void doApplicationBegin(
+                long traceId,
+                long authorization,
+                long affinity)
+            {
+                assert state == 0;
+                state = MqttState.openingInitial(state);
+
+                this.signalFuture = executor.schedule(publishTimeout, SECONDS, routeId, initialId,
+                        PUBLISH_TIMEOUT_SIGNAL);
+
+                router.setThrottle(initialId, this::onApplicationReply);
+                activePublishers.compute(topicFilter, (key, value) -> value != null ? value++ : 1);
```

Comment:
If value is `MutableInteger`, then you can do `computeIfAbsent` followed by `value.value++`.